dotnet/src/SemanticKernel/KernelBuilder.cs

```diff
@@ -77,6 +91,67 @@ public KernelBuilder WithLogger(ILogger log)
         return this;
     }
 
+    /// <summary>
+    /// Enables logging in the kernel to be built.
+    /// </summary>
+    /// <param name="logger">Logger to add.</param>
+    /// <returns>Updated kernel builder with enabled logging.</returns>
+    public KernelBuilder WithLogging(ILogger logger)
+    {
+        Verify.NotNull(logger);
+        this._logger = logger;
+        return this;
+    }
+
+    /// <summary>
+    /// Enables metering in the kernel to be built.
+    /// </summary>
+    /// <param name="meterListener">Instance of <see cref="MeterListener"/> to be used for kernel metering.</param>
+    /// <remarks>
+    /// Note: some properties of passed <see cref="MeterListener"/> will be overridden
+    /// in order to subscribe to all <see cref="Kernel"/> metering events.
+    /// It's recommended to use dedicated instance of <see cref="MeterListener"/> for <see cref="Kernel"/>.
+    /// </remarks>
+    /// <returns>Kernel builder with enabled metering.</returns>
+    public KernelBuilder WithMetering(MeterListener meterListener)
```

Comment:
@noahfalk, @tarekgh, is exposing public API to accept a MeterListener / ActivityListener a pattern we recommend?  This doesn't look like patterns I've seen libraries do before.  Can you comment on the overall design / direction here?

Reply:
I removed methods which accept MeterListener / ActivityListener and provided an example how to configure listeners to listen for SK's diagnostic data on consumer side.